convert-html-entities.js

```diff
@@ -1,7 +1,7 @@
 function convertHTML(str) {
   var charArray = str.split("");
-  for (var i = 0; i < charArray.length; i++){
-    switch (charArray[i]){
+  for (var i = 0; i < charArray.length; i++) {
```

Comment:
You don't need to split strings into arrays, they can be iterated as arrays as they are. An alternative to your code with only strings would be:

```suggestion
let htmlStr = "";
  
  for (var i = 0; i < str.length; i++) {
    switch (str[i]) {
      case "&":
        htmlStr += "&amp;";
        break;
      case "<":
        htmlStr += "&lt;";
        break;
      case ">":
        htmlStr += "&gt;";
        break;
      case '"':
        htmlStr += "&quot;";
        break;
      case "'":
        htmlStr += "&apos;";
        break;
      default:
        htmlStr += str[i]
    }
  }

  return htmlStr;
```

Reply:
Clean and simple. I have to stop splitting every string into arrays...